test/integration/framework/shoot_app.go

```diff
@@ -0,0 +1,550 @@
+	httpRequest, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return err
+	}
+
+	dashboardPassword, err := o.getAdminPassword()
+	if err != nil {
+		return err
+	}
+
+	httpRequest.SetBasicAuth("admin", *dashboardPassword)
+	_, err = httpClient.Do(httpRequest)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// HTTPGet performs an HTTP get with a default timeout of 60 seconds
+func (o *GardenerTestOperation) HTTPGet(url string) (*http.Response, error) {
+	httpClient := http.Client{
+		Timeout: time.Duration(60 * time.Second),
+	}
+
+	httpRequest, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := httpClient.Do(httpRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
+
+// DeployTiller creates a tiller deployment on the shoot cluster
+func (o *GardenerTestOperation) DeployTiller() (*appsv1.Deployment, error) {
+	tillerDeployment, err := o.generateDeploymentObject(o.TillerConfigurationOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	var deploymentToReturn *appsv1.Deployment
+
+	deploymentToReturn, err = o.K8sShootClient.CreateDeploymentObject(tillerDeployment, true)
+	if err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			deploymentToReturn, err = o.K8sShootClient.GetDeployment(o.TillerConfigurationOptions.Namespace, o.TillerConfigurationOptions.DeploymentName)
+			return deploymentToReturn, nil
+		}
+		return nil, err
+	}
+	return deploymentToReturn, nil
+}
+
+// WaitForDeploymentToComplete waits for the shoot to be created
+func (o *GardenerTestOperation) WaitForDeploymentToComplete(deployment *appsv1.Deployment, pollInterval, pollTimeout time.Duration) error {
+	if o.TillerConfigurationOptions == nil {
+		return ErrEmptyTillerOptionsConfiguration
+	}
+
+	return wait.PollImmediate(pollInterval, pollTimeout, func() (bool, error) {
+		_, err := o.K8sShootClient.Kubernetes().AppsV1().Deployments(o.TillerConfigurationOptions.Namespace).Get(o.TillerConfigurationOptions.DeploymentName, metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				o.ShootGardenerTest.Logger.Infof("waiting for deployment: %s to be created \n", o.TillerConfigurationOptions.DeploymentName)
+				return false, nil
+			}
+			return false, err
+		}
+
+		// When all the conditions for the deployment status are true it means that the shoot was successfully created
+		if deploymentCreationCompleted(&deployment.Status) {
+			o.ShootGardenerTest.Logger.Infof("deploymend created!")
+			return true, nil
+		}
+		o.ShootGardenerTest.Logger.Info("Waiting for deployment to be created")
+		return false, nil
+	})
+}
+
+// WaitUntilPodIsRunning waits until the pod with <podName> is running
+func (o *GardenerTestOperation) WaitUntilPodIsRunning(podName, podNamespace, seedOrShoot string, pollInterval, pollTimeout time.Duration) error {
```

Comment:
Better change `seedOrShoot` to `client` and type from `string` to `pkg/client/kubernetes.Clientset`. Then you don't need to to this weird switch case.